src/ai/defaultai.cc

```diff
@@ -5664,13 +5663,13 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 			inputs[22] = (basic_economy_established) ? -1 : 1;
 			inputs[23] = (msites_in_constr() > 0) ? 1 : -2;
 			inputs[24] = (msites_in_constr() > 1) ? 1 : -2;
-			inputs[25] = (wood_policy_ != WoodPolicy::kAllowRangers) * 1;
+			inputs[25] = (wood_policy_ != WoodPolicy::kAllowRangers) * -1;
 			if (gametime > Time(90 * 60)) {
-				inputs[26] = (wood_policy_ != WoodPolicy::kAllowRangers) * 1;
+				inputs[26] = (wood_policy_ == WoodPolicy::kAllowRangers) * 1;
 				inputs[27] = (persistent_data->trees_around_cutters < 20) * 1;
 			}
 			if (gametime > Time(45 * 60)) {
```

Comment:
Is this // middle game, afte 90 minutes ?

Reply:
yes after 45 min